mirdata/datasets/jtd.py

```diff
@@ -0,0 +1,722 @@
+│  │  ├─ bass_onsets.csv
+│  │  ├─ beats.csv
+│  │  ├─ ...
+│  ├─ barronk-beautifullove-mrazgrileyb-2009-c87abfa6
+│  ├─ ...
+```
+
+"""
+
+LICENSE_INFO = """
+
+The MIT License (MIT)
+Copyright (c) 2023, Huw Cheston
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the 
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit 
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the 
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE 
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR 
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+
+
+class Track(core.Track):
+    """JTD track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (str): path to audio file
+        onsets_path (str): path to onsets file
+        midi_path (str): path to MIDI file
+        beats_path (str): path to beats file
+        instrument (str): name of the instrument for this track, either "piano", "bass", or "drums"
+
+    Properties:
+        audio(tuple): audio signal and sample rate for the isolated instrument track of this performance
+        musician (str): name of the musician playing the `instrument` on this track
+
+    Cached Properties:
+        beats (BeatData): beat times for this instrument
+        onsets (EventData): onset and offset times
+        midi (NoteData): midi pitches, onset, offset times, and velocities
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=lambda: json.load(open(self.get_path("metadata"), "r")),
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.onsets_path = self.get_path("onsets")
+        self.midi_path = self.get_path("midi")
+        self.beats_path = self.get_path("beats")
+        self.instrument = self._get_instrument()
+
+    def _get_instrument(self) -> Optional[str]:
+        """Helper function to get the name of the instrument for this track from the track ID"""
+        instrument = self.track_id.split("_")[1]
+        return instrument if instrument in ["piano", "bass", "drums"] else None
```

Comment:
I see quite a few cases where you return `None` if there's no match. In some of these cases, it seems like a failure to find a match would indicate an error. Are you sure it's a good idea to silently suppress such errors?